authutils/validate.go

```diff
@@ -0,0 +1,219 @@
+package authutils
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+// `fence`-specific: the `pur` field indicates the purpose for the token, which
+// may also be validated. If used, it must take one of these values.
+var ALLOWED_PURPOSES []string = []string{"id", "access", "refresh", "session", "api_key"}
```

Comment:
Do we need this outside the package? If not should be lower case. Also you can do like var allowed_purposes []string = [...]string{} to force a fixed size array which helps get at least partial array immutability here.

Reply:
I think it would make sense to export, but it's probably not actually necessary.

Reply:
(in the sense that it's "public" information from the package, but it might not end up being used)